File: Count Subarrays with given XOR.java
class Solution {
    public long subarrayXor(int arr[], int k) {
        
       int n = arr.length;
        HashMap<Integer,Integer> map = new HashMap<>();
        int xor=0;
        int ans=0;
        for(int i=0; i<n; i++){
            xor^=arr[i];
            
            
            if(xor==k)
            ans++;
            
            if(map.containsKey(xor^k)){
                ans += map.get(xor^k);
            }
            
            map.put(xor, map.getOrDefault(xor,0)+1);
        }
        return ans;
    }
}

File: Count Subarrays with given XOR.cpp
class Solution {
  public:
    long subarrayXor(vector<int> &arr, int k) {
        // code here
        int n = size(arr);
        unordered_map<int,int> mp;
        int preXor = 0, count = 0;
        for(int i = 0; i < n; i++){
            preXor ^= arr[i];
            if(preXor == k) 
                count++;
            if(mp.find(preXor ^ k) != mp.end())
                count += mp[preXor ^ k];
            mp[preXor]++;
        }
        return count;
    }
};
